src/__tests__/integration-test.ts

```diff
@@ -102,7 +98,7 @@ class MockTracer {
 
     return {
       log(object) {
-        self.spans.find(span => span.name === name).logs.push(object);
+        self.spans.find(span => span.id === spanId).logs.push(object);
```

Comment:
There was a bug in this logic that broke for array tests but went unnoticed in the old snapshots.